auth_proxy/proxy/flask.py

```diff
@@ -34,7 +52,27 @@ def __init__(self, url, orig):
     def request(self):
         """ @inherit
         """
-        url = self.url + '?' + parse.urlencode(self.orig.args)
+        args = dict(self.orig.args)
+
+        # Get the authorized scopes from the request, or open everything if
+        # this is an un-authorized request
+        try:
+            scopes = list(self.orig.oauth.scopes)
+        except AttributeError:
+            scopes = ['patient/*.read',]
+
+        # Wildcard scopes should be expanded
+        try:
+            wildcard = scopes.index('patient/*.read')
+            scopes[wildcard:wildcard] = self.ccds_scopes
+        except ValueError:
+            pass
+
+        # Update the query params
+        args['_security'] = ' '.join(scopes + ['public'] + args.get('_security', ''))
```

Comment:
This logic doesn't look quite right to me. The end state we want is something like:

   &_security=public,patient,problems&_security=public,Patient/123

Note that this means:

 * Repeating `security` parameters, one controlling the data categories and one controlling the patient record(s)
 * `public` appears in each list to ensure that `public` resources can be ready by anyone
 * `_security` from the incoming request is summarily dropped; we don't want our proxy to propagate this through (since we don't want the requester to be responsible for telling us what they're allowed to see).

FHIR's search logic says each available resource must have "one value from each comma-separated list"